ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java

```diff
@@ -51,6 +53,17 @@ public void handleStatusChanged(TableRuntime tableRuntime, OptimizingStatus orig
     tryEvaluatingPendingInput(tableRuntime, loadTable(tableRuntime));
   }
 
+  @Override
+  public void handleConfigChanged(TableRuntime tableRuntime, TableConfiguration originalConfig) {
```

Comment:
This can solve the problem that Optimizing cannot be triggered when self-optimizing.enabled changes from false to true, but I think this part should be abstracted. There should be only one method call for tryEvaluatingPendingInput? A state should be maintained when handleConfigChanged is triggered, and it will be found in the next TableRuntimeRefresh And trigger, but unfortunately, the current TableRuntime cannot support this judgment

Reply:
I agree that we should limit the method `tryEvaluatingPendingInput` only be called in the executor thread pool and I found that `TableRuntimeRefreshExecutor#handleStatusChanged` had called the method already now which should be fixed too.

`BaseTableExecutor#scheduleIfNecessary` is used when you want to trigger a function on a table proactively and I think we should call it in `handleStatusChanged` and `handleConfigChanged`.

The problem is that when the current snapshot of the table is not changed, `tryEvaluatingPendingInput` will be skipped, and we may need to force the evaluating process in some way, like setting the `currentSnapshotId` to -1 or add a non-persistent variable to store this state.

Reply:
I think setting `currentSnapshotId` to -1 is not a good option. 

We can introduce an additional field, `lastOptimizedSnapshotId`, to record the snapshotId when optimizing was last executed.

Reply:
1. I agree that refresh operation should only happen in execute() method
2. handleStatusChanged() or handleConfigChanged() could only call scheduleIfNecessary()
3. enabled() is the key for RefreshingExecutor to determine whether a table should be refreshed. I don't think current impl is a good idea:
`  
@Override
  protected boolean enabled(TableRuntime tableRuntime) {
    return  loadTable(tableRuntime).originalTable() instanceof ArcticTable;
  }
`